sql/core/src/test/scala/org/apache/spark/sql/SQLContextSuite.scala

```diff
@@ -51,7 +51,7 @@ class SQLContextSuite extends SparkFunSuite with SharedSparkContext {
 
     // UDF should not be shared
     def myadd(a: Int, b: Int): Int = a + b
-    session1.udf.register[Int, Int, Int]("myadd", myadd)
+    session1.udf.register[Int, Int, Int]("myadd", myadd(_, _))
```

Comment:
Why was this change needed? Is this a breaking change for existing code?

Reply:
I'm glad you asked. Consider the signature of the new register overloading:

```scala
  def register[IN: TypeTag, BUF, OUT](
      name: String,
      agg: Aggregator[IN, BUF, OUT]): UserDefinedAggregator[IN, BUF, OUT] = {
```

If you don't actually scala you are sending it something function-like, it will now be confused in the Function2[] case

Reply:
So, in that particular Function2[] case, I guess it is a breaking change. I felt that was acceptable since we're looking at spark 3.0

Reply:
Not sure I follow why it is confusing Scala. Why would adding a new method that accepts Aggregator cause another overload to stop working? Aggregator isn't a Function2

Reply:
Good question. Scala type inference is a wonderland.  I'm pretty sure it isn't a coincidence that it was the 3-type-parameter overloading that got confused, tho.

Reply:
So is the problem that this has 3 type parameters, so the method can't be automatically converted to `Function2[R, A1, A2]`? That doesn't make much sense to me since a method clearly isn't an `Aggregator[IN, BUF, OUT]`

Reply:
My bet is some interaction of the default lifting of `myadd` into `(Int,Int)=>Int`, combined with type erasure.

Reply:
to make our life easier, how about we use `registerAggregator` as the new function name?

Reply:
I'm ok with a new method name if everyone else is

Reply:
It would be nice to know what's going on here with Scala, but I don't think this is worth breaking existing code (even if we _can_ for 3.0). Unless we find out what the underlying problem is, I think we should use `registerAggregator`.